content/blog/2020/03/2020-03-08-IntelliJ-Setup-for-Jenkins-Core-Development.adoc

```diff
@@ -0,0 +1,107 @@
+
+Index:
+
+1. Installation and Cloning
+2. Setting up IntelliJ IDEA
+3. Building Jenkins
+4. Debugging
+
+== 1. Installation and Cloning
+
+Install IntelliJ from link:https://www.jetbrains.com/help/idea/installation-guide.html[here], and clone the latest Jenkins core source code from link:https://github.com/jenkinsci/jenkins[here] (you can also fork it before cloning). In this tutorial, I have used IntelliJ IDEA Version 2019.3.1.
+
+ git clone https://github.com/jenkinsci/jenkins
+
+== 2. Setting up IntelliJ IDEA
+
+Open IntelliJ IDEA.
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup1.jpg[]
+
+Click on Import Project, and in the popup, choose the directory where Jenkins had been cloned.
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup2.jpg[]
+
+Select ‘Create project from existing sources’, and click on Next.
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup3.jpg[]
+
+Click on Next
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup4.jpg[]
+
+Click on Next
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup5.jpg[]
+
+Click on Next
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup6.jpg[]
+
+Click on Finish. IntelliJ should be ready like below.
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/setup7.jpg[]
+
+== 3. Building Jenkins
+
+Now we want to build Jenkins. Run the following: (This will prepare the jenkins.war file without running tests)
+
+ mvn -am -pl war,bom -DskipTests -Dspotbugs.skip clean install
+
+This may take a little time. If everything works well, you should get a successful build result like below:
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/build1.jpg[]
+
+Now, we can run Jenkins on our local machine using the following command:
+
+ mvn -pl war jetty:run
+
+If it runs successfully, you should be able to view Jenkins console by going to http://localhost:8080/jenkins/
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/build2.jpg[]
+
+== 4. Debugging
+
+Now, our next goal is to be able to debug jenkins by using breakpoints in IntelliJ. Close the previous running instance by (Ctr+c).
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/debug1.jpg[]
+
+In the toolbar, go to Run -> Edit Configurations…
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/debug2.jpg[]
+
+Click on the ‘+’ icon so we can add a new configuration.
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/debug3.jpg[]
+
+Click on ‘Maven’.
+
+image::/images/post-images/2020/03-IntelliJ-Setup-for-Jenkins-Core-Development/debug4.jpg[]
+
+Ensure that this configuration as same as above. Click on Apply and OK.
```

Comment:
This `Ensure that this configuration as same as above` should be `Ensure that this configuration is the same as above'